### ISSUE TITLE
Identification of FOPDT Systems

### ISSUE THREAD

**Issue by bykan** [dynamicslab/pysindy#105]
I tried to use PySINDy to identify a simple first-order linear system with time delay. [The dataset](http://apmonitor.com/do/uploads/Main/tclab_siso_data.txt) is composed of control input (Voltage) and the measured output (Temperature) as shown below. 

![image](https://user-images.githubusercontent.com/41585949/113434961-e5450e00-93af-11eb-96ba-2beda9bed47d.png)

I used the following code snippet to identify the dynamics of the system:

`x_train = data['temperature'].values`
`u_train = data['voltage'].values`
`t_train = data['time'].values`
`dt = 1`

`# Instantiate and fit the SINDYc model`

`stlsq_optimizer = ps.STLSQ(threshold=1e-6, alpha=0.01, max_iter= 100, normalize= True)`
`poly_library = ps.PolynomialLibrary(include_interaction=False, include_bias=False, degree= 2)`

`model = ps.SINDy(optimizer=stlsq_optimizer, feature_library= poly_library, discrete_time=False)`
`model.fit(x_train, u=u_train, t=1)`
`model.print()`

and I got the following result:
`x0' = 0.003 x0 + 0.001 u0`

and the model score is 
`# Compare SINDy-predicted derivatives with finite difference derivatives`
`print('Model score: %f' % model.score(x_train, u=u_train, t=t_train))`

`Model score: 0.341724`

Can you please show me how to improve the performance and boost the model score?

**Comment by akaptano** [comment]
Hi bykan,

Could you please show an illustration of the true data and fitted model of x_train and d/dt (x_train) together? Also, please use model.print(precision=6) or something like that, so we can see if there are some smaller coefficients in the model. My guess, before seeing the model fit, is that either the threshold is too big (What does it look like when threshold=0?) or the library is not sufficient to capture the data (my understanding is that a quadratic library with just T and V is [T, V, T**2, TV, V**2]). Is there an analytic model or reason to think this library is sufficient? 

Best,
Alan Kaptanoglu

**Comment by bykan** [comment]
> Hi bykan,
> 
> Could you please show an illustration of the true data and fitted model of x_train and d/dt (x_train) together? Also, please use model.print(precision=6) or something like that, so we can see if there are some smaller coefficients in the model. My guess, before seeing the model fit, is that either the threshold is too big (What does it look like when threshold=0?) or the library is not sufficient to capture the data (my understanding is that a quadratic library with just T and V is [T, V, T**2, TV, V**2]). Is there an analytic model or reason to think this library is sufficient?
> 
> Best,
> Alan Kaptanoglu

Thanks, Alan for your quick response. The mission is accomplished simply by identifying the model as a discrete-time.
For a first-order model, the fitted model against true data:

![image](https://user-images.githubusercontent.com/41585949/113458725-3b7e7500-93e1-11eb-8f2a-a720294c4aee.png)

For a second-order model:

![image](https://user-images.githubusercontent.com/41585949/113458750-581aad00-93e1-11eb-9b5e-4f01d7c901b8.png)

Model score: 0.999755

**Comment by bykan** [comment]
Is it possible to identify a continuous-time model for the same data?

**Comment by akaptano** [comment]
The discrete model looks pretty good. I don't see why you can't do the same for a continuous-time model, which is what I thought you were doing earlier, since you used "discrete_time=False" in the model. Again, it would be helpful if you outputted the model coefficients and plotted both x_train (true and fit) and the derivative of x_train (true and fit).

Best,
Alan

**Comment by bykan** [comment]
> The discrete model looks pretty good. I don't see why you can't do the same for a continuous-time model, which is what I thought you were doing earlier, since you used "discrete_time=False" in the model. Again, it would be helpful if you outputted the model coefficients and plotted both x_train (true and fit) and the derivative of x_train (true and fit).
> 
> Best,
> Alan

Thanks, Alan for your quick feedback. There was a bug in my code for simulating the behaviour of the continuous-time model. The problem is now solved and there's an excellent match between the measured data and the fitted second-order model. However, The model score is still 0.341724 for a 2nd order continuous-time model vs  0.999755 for a 2nd order discrete-time model.

![image](https://user-images.githubusercontent.com/41585949/113463238-bdc36500-93f2-11eb-8195-79a132afe6ff.png)
